packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -200,6 +206,17 @@ const SceneView = ({
     freezeOnBlur,
   } = options;
 
+  const screenRef = React.useRef(null);
+  React.useEffect(() => {
```

Comment:
Layout effect would be better since ref is assigned earlier.

```suggestion
  React.useLayoutEffect(() => {
```

Reply:
Done in https://github.com/react-navigation/react-navigation/pull/11943/commits/301cd8dace25c46ed8b0f5931333176b07dcc99c.